File: leetcode/0021-merge-two-sorted-lists/solution.cpp
class Solution {
    public:
        ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
            ListNode* dummy = new ListNode(0);
            ListNode* current = dummy;
    
            while (list1 && list2) {
                if (list1 -> val > list2 -> val) {
                    current -> next = list2;
                    list2 = list2 -> next;
                } else {
                    current -> next = list1;
                    list1 = list1 -> next;
                }
                current = current -> next;
            }
            current -> next = list1 ? list1 : list2;
            
            ListNode* newList = dummy -> next;
            delete dummy;
            return newList;
        }
};

File: leetcode/0021-merge-two-sorted-lists/solution.py
class Solution(object):
    def mergeTwoLists(self, list1, list2):
        dummy = ListNode()
        current = dummy

        while list1 and list2:
            if list1.val > list2.val:
                current.next = list2
                list2 = list2.next
            else:
                current.next = list1
                list1 = list1.next
            
            current = current.next
        
        if list1:
            current.next = list1
        else:
            current.next = list2
        
        return dummy.next